### ISSUE TITLE
CephObjectRealm's endpoint field validation issues 

### ISSUE THREAD

**Issue by libmonsoon-dev** [rook/rook#9835]
After updating rook to v1.8.5, when creating a CephObjectRealm, I get an error:

`CephObjectRealm.ceph.rook.io "object-store" is invalid: spec.pull.endpoint: Invalid value: "": spec.pull.endpoint in body should match '^https*://'`

CephObjectRealm is created with the following code:
```
import (
	"context"
	"fmt"

	cephrookiov1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
	cephv1 "github.com/rook/rook/pkg/client/clientset/versioned/typed/ceph.rook.io/v1"
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
)

func CreateCephObjectRealm(ctx context.Context, cephclient cephv1.CephV1Interface, name, namespace string) error {
	realm := &cephrookiov1.CephObjectRealm{
		ObjectMeta: metav1.ObjectMeta{
			Name:      name,
			Namespace: namespace,
		},
		TypeMeta: metav1.TypeMeta{
			APIVersion: "ceph.rook.io/v1",
			Kind:       "CephObjectRealm",
		},
	}

	_, err := cephclient.CephObjectRealms(namespace).Create(ctx, realm, metav1.CreateOptions{})
	if err != nil {
		return fmt.Errorf("create ceph object realm: %w", err)
	}
	
	return nil
}
```

This worked in earlier versions, but stopped after an update.
I think it's a matter of serialization of structures in golang and this field will become optional again only if CephObjectRealm.Spec is stored as a pointer to the ObjectRealmSpec structure

**Comment by travisn** [comment]
Looks like we need to add the following to the [pullSpec.Endpoint](https://github.com/rook/rook/blob/release-1.8/pkg/apis/ceph.rook.io/v1/types.go#L1553):
```
	// +optional
```
@alimaredia Could you take a look? thanks

**Comment by libmonsoon-dev** [comment]
We already have +optional directive on [CephObjectRealm.Spec](https://github.com/rook/rook/blob/release-1.8/pkg/apis/ceph.rook.io/v1/types.go#L1531), but it doesn't seem to work

**Comment by libmonsoon-dev** [comment]
By the way, I already tried to generate manifests in the fork with an optional endpoint field, it did not work, the error remained the same 

> Looks like we need to add the following to the [pullSpec.Endpoint](https://github.com/rook/rook/blob/release-1.8/pkg/apis/ceph.rook.io/v1/types.go#L1553):
> 
> ```
> 	// +optional
> ```

**Comment by travisn** [comment]
@libmonsoon-dev As a workaround, can you just remove the validation [here](https://github.com/rook/rook/blob/master/pkg/apis/ceph.rook.io/v1/types.go#L1558) and generate the crd again? Or just remove the url validation from the crds.yaml you create.

**Comment by travisn** [comment]
This was fixed by #10139
